### PR TITLE
dev: Add dark mode support for Layout

### PR DESCRIPTION
**Pull request by jcbcapps**
# SC-1027

## Proposed changes
Adds dark mode support for the following components:
- PageNav
- PageHeader
- Header
- Footer
- EditDisplayName

---

## Reviewer notes
View each component while logged into the portal and verify that styles are accurate for both light and dark mode. You can also view each component in Storybook.

## Setup
Make sure to have both the CMS and the portal client running on your local machine. You can do this by
- Running `yarn services:up && yarn dev` in the CMS repo and
- Running `yarn dev` in the portal client with this branch checked out

To toggle between light and dark mode, open the dev tools in your browser when running the portal, then click Application > Local Storage > Set a key/value pair for theme, dark.

---

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [ ] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
  - [ ] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On commercial internet in Firefox
  - [ ] On commercial internet in Chrome
  - [ ] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

### CODE REVIEW

File: src/components/EditDisplayName/EditDisplayName.module.scss [hunk 1]
```diff
@@ -22,3 +22,10 @@
 .successMessage {
   margin-top: 8px;
 }
+
+.saveDisplayName {
+  &:hover:disabled {
+    cursor: default;
+    background-color: var(--btn-primary-bg);
+  }
```
Comment (jcbcapps) [inline src/components/EditDisplayName/EditDisplayName.module.scss:30]: I noticed that the `Save` button was disabled but still changing the cursor and bg color on hover so I updated it!

**Review by gidjin — COMMENTED** [review]
Question: When I load the page locally I see a white bar at the bottom before the footer. Is that to do with the overall background or another component? Is fixing that covered in this story or is that a later story?

Screenshot
SCR-20220926-g2f

**Comment by abbyoung** [comment]
A couple changes from reviewing Happo!

* [Background of default footer](https://happo.io/a/909/p/1321/compare/6baa66a10fd75bde9c0cdde72cb06770bdd3d4d2/557ac86d9ac4e4b6e7effbc193cb70f6dbe8c69a#Layout%5C%2FFooter-Default%5C%20Footer)
Screen Shot 2022-09-29 at 5 15 39 PM

* Very hard to see, but the color of the active state in the side page nav is a slightly different color -- [link to happo](https://happo.io/a/909/p/1321/compare/6baa66a10fd75bde9c0cdde72cb06770bdd3d4d2/557ac86d9ac4e4b6e7effbc193cb70f6dbe8c69a#Components%5C%2FPageNav-On%5C%20Sites%5C%20And%5C%20Applications%5C%20Page)

* It looks like the header on News & Announcements and Portal home is getting wiped out / missing the bg image - [link to happo](https://happo.io/a/909/p/1321/compare/6baa66a10fd75bde9c0cdde72cb06770bdd3d4d2/557ac86d9ac4e4b6e7effbc193cb70f6dbe8c69a#Components%5C%2FPageNav-On%5C%20Sites%5C%20And%5C%20Applications%5C%20Page) I'm not sure if that was intentional / are we removing the image? Or just for dark mode?

**Comment by jcbcapps** [comment]
> A couple changes from reviewing Happo! If any of these are expected + should be approved, please let me know. Thank you!

1. I did intentionally make the change for the color of the footer. I noticed that it was black in both light and dark mode designs.
2. Good catch! Took me a minute of toggling it to even notice. I'll update it!
3. It was intentional but I was thinking it looked weird in light mode and wanted to discuss with design. I'll ask @malworks and make any necessary changes.

**Comment by abbyoung** [comment]
One more thing 🙏 Could you take a look at the logout button in the header? I think it needs some updating for dark mode.

File: src/styles/_themes.scss [hunk 2]
```diff
@@ -122,9 +136,17 @@
...
+      180deg,
+      #000000 6.36%,
+      #0f305a 73.55%,
+      #555f92 111.89%
+    )};
+
   --body-bg: #{$theme-spacebase-dark};
   --bg: #{$theme-spacebase-dark};
   --text: #{$white};
+  --header-image: none;
```
Comment (gidjin) [inline src/styles/_themes.scss:149]: thought: this looks strange to me, but confirmed it is what is in figma. Probably waiting for a dark version of the header image.

**Review by gidjin — APPROVED** [review]
LGTM with the changes!

**Review by abbyoung — APPROVED** [review]

**Pull request merged**